torchgeo/models/rcf.py

```diff
@@ -104,3 +107,111 @@ def forward(self, x: Tensor) -> Tensor:
+    model is instantiated with a dataset, samples patches from the dataset, ZCA whitens
+    the patches, then uses those as convolutional filters to extract features with.
+
+    .. note::
+
+        This Module is *not* trainable. It is only used as a feature extractor.
+    """
+
+    def _normalize(
+        self,
+        patches: "np.typing.NDArray[np.float32]",
+        min_divisor: float = 1e-8,
+        zca_bias: float = 0.001,
+    ) -> "np.typing.NDArray[np.float32]":
+        """Does ZCA whitening on a set of input patches.
+
+        Copied from https://github.com/Global-Policy-Lab/mosaiks-paper/blob/7efb09ed455505562d6bb04c2aaa242ef59f0a82/code/mosaiks/featurization.py#L120
+
+        Args:
+            patches: a numpy array of size (N, C, H, W)
+            min_divisor: a small number to guard against division by zero
+            zca_bias: bias term for ZCA whitening
+
+        Returns
+            a numpy array of size (N, C, H, W) containing the normalized patches
+        """  # noqa: E501
+        n_patches = patches.shape[0]
+        orig_shape = patches.shape
+        patches = patches.reshape(patches.shape[0], -1)
+
+        # Zero mean every feature
+        patches = patches - np.mean(patches, axis=1, keepdims=True)
+
+        # Normalize
+        patch_norms = np.linalg.norm(patches, axis=1)
+
+        # Get rid of really small norms
+        patch_norms[np.where(patch_norms < min_divisor)] = 1
+
+        # Make features unit norm
+        patches = patches / patch_norms[:, np.newaxis]
+
+        patchesCovMat = 1.0 / n_patches * patches.T.dot(patches)
+
+        (E, V) = np.linalg.eig(patchesCovMat)
+
+        E += zca_bias
+        sqrt_zca_eigs = np.sqrt(E)
+        inv_sqrt_zca_eigs = np.diag(np.power(sqrt_zca_eigs, -1))
+        global_ZCA = V.dot(inv_sqrt_zca_eigs).dot(V.T)
+        patches_normalized: "np.typing.NDArray[np.float32]" = (
+            (patches).dot(global_ZCA).dot(global_ZCA.T)
+        )
+
+        return patches_normalized.reshape(orig_shape).astype("float32")
+
+    def __init__(
+        self,
+        dataset: NonGeoDataset,
+        in_channels: int = 4,
+        features: int = 16,
+        kernel_size: int = 3,
+        bias: float = -1.0,
+        seed: Optional[int] = None,
+    ):
+        """Initializes the MOSAIKS model.
+
+        This is the main model used in Multi-task Observation using Satellite Imagery
+        & Kitchen Sinks (MOSAIKS) method proposed in
+        https://www.nature.com/articles/s41467-021-24638-z.
+
+        Args:
+            dataset: a torch dataset that returns dictionaries with an "image" key
+            in_channels: number of input channels
+            features: number of features to compute, must be divisible by 2
+            kernel_size: size of the kernel used to compute the RCFs
+            bias: bias of the convolutional layer
+            seed: random seed used to initialize the convolutional layer
+
+        .. versionadded:: 0.5.0
```

Comment:
```suggestion
        .. versionadded:: 0.5
```
Also, this should be in the class docstring, not the `__init__` docstring. The entire class is new, not just the `__init__` method.